static/js/components/LearnerInfoCard.js

```diff
@@ -18,50 +22,85 @@ export default class LearnerInfoCard extends React.Component {
     profile: Profile,
     toggleShowPersonalDialog: () => void,
     toggleShowAboutMeDialog: () => void,
+    openLearnerEmailComposer: () => void,
   };
 
-  email = (email: string): React$Element<*> => (
-    <span className="profile-email">{email}</span>
-  );
+  ownProfilePage: boolean;
+
+  isOwnProfilePage = (): boolean => {
+    if (R.isNil(this.ownProfilePage)) {
+      this.ownProfilePage = isProfileOfLoggedinUser(this.props.profile);
+    }
+    return this.ownProfilePage;
+  };
```

Comment:
I think in general if you wanted to avoid extra computationally expensive calculation you would store the calculated value in state and look it up via some key in the props. There's also `shouldComponentUpdate` which can be used to skip rerenders if it's obvious they're not needed (for example if your props were entirely immutable and comparing the references would be enough to ensure equality)